url47af6b1535_rolandotr_db_comparison_tgz-pJ8-methodology_HistoryJ8/src/utils/Gnuplot.java

```diff
@@ -0,0 +1,49 @@
+package utils;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Hashtable;
+
+import protocols.specifications.DBProtocol;
+
+public abstract class Gnuplot {
+
+	
+	public static void main(String[] args) {
+		
+	}
+	
+	public static void createDatafile(DBProtocol[] protocols,
+			DBProtocol[] frontiers, String preffix) throws IOException {
+		String newLine = System.getProperty("line.separator");
+		File points = new File("points_"+preffix+".DAT");
+		FileWriter writer = new FileWriter(points);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *writer*. The resource is referenced at lines 23, 24, 25, 26. The resource is closed at line 28. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *writer* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).